src/oci/image/mod.rs

```diff
@@ -0,0 +1,84 @@
+pub mod config;
+pub mod layer;
+pub mod manifest;
+
+pub trait Reference {
+    fn registry(self) -> String;
+
+    fn fullname(self) -> String;
+
+    fn tag(self) -> String;
+
+    fn digest(self) -> String;
+}
+
+#[derive(Debug, PartialEq)]
+pub struct ImageReference(String);
+
+impl Reference for ImageReference {
+    fn registry(self) -> String {
+        match self.0.find("/") {
+            Some(x) => self.0[..x].to_string(),
+            None => "docker.io".to_string(),
+        }
+    }
+
+    fn fullname(self) -> String {
+        match self.0.find("/") {
+            Some(x) => self.0[x + 1..].to_string(),
```

Comment:
Can't you use "into" instead?

Reply:
I'm not sure to understand the difference, might read things on it before.

Reply:
From what I read [here](https://users.rust-lang.org/t/what-is-the-idiomatic-way-to-convert-str-to-string/12160/8) `into` would work but would be preferred when converting from another type.
Not sure what to choose though...